File: my-solutions/0049-group-anagrams/solution.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string, vector<string>> mp;

        for (const auto& x : strs) {
            string word = x;

            sort(word.begin(), word.end());

            mp[word].push_back(x);
        }

        vector<vector<string>> ans;
        ans.reserve(mp.size());
        for (auto& entry : mp) {
            ans.push_back(std::move(entry.second));
        }
        return ans;
    }
};

File: my-solutions/0049-group-anagrams/solution.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        groups = defaultdict(list)
        for word in strs:
            key = [0] * 26
            for c in word:
                key[ord(c) - ord('a')] += 1
            groups[tuple(key)].append(word)
        return list(groups.values())
